typing/typecore.ml

```diff
@@ -856,24 +891,9 @@ let disambiguate_lid_a_list loc closed env expected_type lid_a_list =
        * if the reduced list is valid, call Label.disambiguate
      *)
     let scope = Env.lookup_all_labels ~loc:lid.loc lid.txt env in
-    match expected_type, scope with
-    | None, Error(loc, env, err) ->
-        Env.lookup_error loc env err
-    | Some _, Error _ ->
-        Label.disambiguate () lid env expected_type scope ~warn ~scope
-    | _, Ok lbls ->
-       let disambiguate_labels lbls =
-         Label.disambiguate () lid env expected_type (Ok lbls) ~warn ~scope in
-       match disambiguate_label_by_ids closed ids lbls with
-         | Ok lbls -> disambiguate_labels lbls
-         | Error lbls ->
-            if (expected_type <> None)
-            (* both branches will fail later; if there an expected type,
-               it will be used for a nice error message; otherwise we return
-               an arbitrary candidate for use in best-effort error messages. *)
-            then disambiguate_labels []
-            else fst (List.hd lbls)
```

Comment:
The test failure is due to a small change of behavior in this branch since we never reached the `warn_if_disambiguated` before the PR.

Reply:
Yes, I updated the test.